### PR TITLE
Bleve server

### PR DESCRIPTION
**Pull request by timburks**
This adds a `bleve serve` subcommand which runs an experimental http server providing [bleve](https://blevesearch.com)-based API search.

See the README for usage. So far it's been used standalone locally and in a Kubernetes cluster using a `PersistentVolumeClaim` for index storage. 

It's intended for use in containers run alongside `registry-server` in clusters and other environments.

New tests verify that the server APIs are callable but only minimally verify the search and indexing operations.

### CODE REVIEW

File: cmd/registry-experimental/cmd/bleve/bleve_test.go [hunk 1]
```diff
@@ -181,3 +191,83 @@ func TestSearch(t *testing.T) {
...
+	go func() {
+		cmd := Command()
+		cmd.SetArgs([]string{"serve", "--bleve", blevePath, "--port", port})
+		if err := cmd.Execute(); err != nil {
+			log.Fatalf("Execute() with args %+v returned error: %s", cmd.Args, err)
+		}
+	}()
+
+	// Give it time to start.
+	time.Sleep(1 * time.Second)
```
Comment (theganyo) [inline cmd/registry-experimental/cmd/bleve/bleve_test.go:212]: If possible, we should avoid non-deterministic stuff that could flake a test. I know we don't have a health check, but maybe we could just do `net.DialTimeout` in a bounded loop?

**Review by theganyo — APPROVED** [review]

File: cmd/registry-experimental/cmd/bleve/bleve_test.go (hunk 1 above)
Comment (timburks) [reply cmd/registry-experimental/cmd/bleve/bleve_test.go:212]: OK, thanks! I'll fix that before submitting

File: cmd/registry-experimental/cmd/bleve/bleve_test.go [hunk 2]
```diff
@@ -181,3 +191,83 @@ func TestSearch(t *testing.T) {
...
+
+	// Call the indexing API.
+	{
```
[the remark this answers is not in the document]
Comment (timburks) [reply cmd/registry-experimental/cmd/bleve/bleve_test.go:215]: ack

File: cmd/registry-experimental/cmd/bleve/bleve_test.go [hunk 3]
```diff
@@ -181,3 +192,86 @@ func TestSearch(t *testing.T) {
...
+	go func() {
+		cmd := Command()
+		cmd.SetArgs([]string{"serve", "--bleve", blevePath, "--port", port})
+		if err := cmd.Execute(); err != nil {
+			log.Fatalf("Execute() with args %+v returned error: %s", cmd.Args, err)
+		}
+	}()
+
+	// Wait for the server to start.
+	_, err := net.DialTimeout("tcp", "localhost:"+port, 2*time.Second)
```
Comment (timburks) [inline cmd/registry-experimental/cmd/bleve/bleve_test.go:213]: @theganyo here's the update to use net.DialTimeout() - thanks, this is much better!

**Pull request merged**